src/main/java/core/basesyntax/service/impl/MyFileReaderImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.FileReader;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MyFileReaderImpl implements FileReader {
+    @Override
+    public List<String> read(String filePath) {
+        List<String> lines = new ArrayList<>();
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream(filePath);
+        if (inputStream == null) {
+            throw new RuntimeException("File not found: " + filePath);
+        }
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(inputStream))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading file: " + filePath, e);
+        }
```

Comment:
Good use of try-with-resources to ensure that the BufferedReader is closed properly. This adheres to the checklist item regarding the use of AutoCloseable resources.